### ISSUE TITLE
TypeError: Failed to execute 'fetch'  in Blazor Webassembly

### ISSUE THREAD

**Issue by fabiohcnobre** [restsharp/RestSharp#2248]
I am using the code below

``` csharp
var options = new RestClientOptions("http://apibihml.value4u.com.br");
var client = new RestClient(options);
var request = new RestRequest("/obter-acesso/", Method.Get);
request.AddHeader("Host", "apibihml.value4u.com.br");
request.AddHeader("Content-Type", "application/json; charset=utf-8");
request.AddHeader("AcceptCharset", "utf-8");
RestResponse response = await client.ExecuteAsync(request);
Console.WriteLine(response.Content);
```
Erro in blazor
 
▶ | ErrorMessage | 'TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.' | 

![image](https://github.com/user-attachments/assets/f9e12ed1-b911-419c-a411-fc2d478232d4)

![image](https://github.com/user-attachments/assets/056427a0-92ff-4c85-a113-dbaf295fb94a)

 - OS: Windows 
 - .NET version 8
 -  Blazor  Webassembly

**Comment by alexeyzimarev** [comment]
Yes, GET requests cannot have a body. Why are you setting the content type header if your request has no content???

**Comment by fabiohcnobre** [comment]
Even without a header it gives an error.

![image](https://github.com/user-attachments/assets/921af598-74b4-431b-a722-b79c567a0cf6)

I created this repository for testing.
https://github.com/fabiohcnobre/TestBlazor.BI.Web

**Comment by alexeyzimarev** [comment]
I am wondering if you read the docs https://restsharp.dev/docs/usage/client#blazor-support

**Comment by fabiohcnobre** [comment]
Yes,
Same error using the information. You can see the error in the repository below.
https://github.com/fabiohcnobre/TestBlazor.BI.Web

**Comment by alexeyzimarev** [comment]
Sorry, but I don't really understand you. Your repository has this code _on the page_:

```csharp
            var options = new RestClientOptions("https://apibihml.value4u.com.br");
            var client = new RestClient(options);
```

The documentation clearly says that you need to register the client using the Blazor version of `HttpClient`:

```csharp
builder.Services.AddSingleton(new RestClient(new HttpClient()));
```

followed by a page example where you use the injected client.

The docs additionally points out that:

> You need to remember that webassembly has some platform-specific limitations. Therefore, you won't be able to instantiate `RestClient` using all of its constructors. In fact, you can only use `RestClient` constructors that accept `HttpClient` or `HttpMessageHandler` as an argument. 

I don't see any of this in your repository.

**Comment by alexeyzimarev** [comment]
In addition, calling the API in Postman doesn't really help. Blazor HttpClient uses the fetch function, and the best way to understand what's going on is to open the dev tools in the browser and look at both console and network calls tabs. You'll see if the request actually has a body, what the server returns (if anything), or if the fetch hasn't been executed, as well as the error message from the fetch call.

**Comment by fabiohcnobre** [comment]
@alexeyzimarev 

I hadn't updated github because it gave the same error. But I updated the repository and the same error continues.


![image](https://github.com/user-attachments/assets/a4c3ec40-aabc-46a3-b379-e2c9d4d622ba)
